src/server/tiering/common.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <memory>
+#include <optional>
+
+#include "util/fibers/synchronization.h"
+
+namespace dfly::tiering {
+
+// Location on the offloaded blob, measured in bytes
+struct DiskSegment {
+  DiskSegment FillPages() const {
+    return {offset / 4096 * 4096, (length + 4096 - 1) / 4096 * 4096};
+  }
```

Comment:
In a separate PR I'll move external_alloc and io_mgr to tiering subfolder and make them reuse the kb literal, so we don't have separate 4096/4_kb notations